resolver/interface.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resolver
+
+import (
+	"context"
+
+	servicebindingv1alpha3 "github.com/servicebinding/service-binding-controller/apis/v1alpha3"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Resolver interface {
+	// LookupMapping returns the mapping template for the workload. Typically a ClusterWorkloadResourceMapping is defined for the workload's
+	// fully qualified resource `{resource}.{group}`. The workload's version is either directly matched, or the wildcard version `*`
+	// mapping template is returned. If no explicit mapping is found, a mapping appropriate for a PodSpecable resource may be used.
+	LookupMapping(ctx context.Context, workload client.Object) (*servicebindingv1alpha3.ClusterWorkloadResourceMappingTemplate, error)
```

Comment:
how about referring here just `MappingSource` interface?

```golang
type Resolver interface {
  MappingSource
  LookupBindingSecret(ctx context.Context, serviceRef corev1.ObjectReference) (string, error)
  LookupWorkload(ctx context.Context, workloadRef corev1.ObjectReference) (client.Object, error)
}
```

Reply:
I was trying to avoid creating a dependency between the two packages. If you think it's important to share a common interface type, we can pull them into a third package.